### PR TITLE
Add Kotlin Client

### PR DESCRIPTION
**Pull request by krisbitney**
Closes https://github.com/polywrap/kotlin-client/issues/30

Note that I made the URIs explicit strings in the yaml spec files. This is required for yaml parsing.

Also, there is a filter set for the Kotlin client CI. Tests are only run if they don't require `tryResolveUri` or `loadWrapper`, since those methods aren't currently supported in the FFI.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,48 +3,48 @@ A place to define a "readiness checklist" for all Polywrap clients to adhere to.
...
-| | `Enum` | Yes | [:mag:](./specs/wrap_type_enum.yaml) | [:heavy_check_mark:](./clients/js/src/features/wrap_type_enum.ts) |  
-| | `BigInt` | Yes | [:mag:](./specs/wrap_type_bigint.yaml) | [:heavy_check_mark:](./clients/js/src/features/wrap_type_bigint.ts) |  
-| | `BigNumber` | Yes | [:mag:](./specs/wrap_type_bignumber.yaml) | [:heavy_check_mark:](./clients/js/src/features/wrap_type_bignumber.ts) |  
-| | `JSON` | Yes | [:mag:](./specs/wrap_type_json.yaml) | [:heavy_check_mark:](./clients/js/src/features/wrap_type_json.ts) |  
-| | `Array<T>` | Yes | TODO | TODO |  
-| | `Map<K, V>` | Yes | [:mag:](./specs/wrap_type_map.yaml) | [:heavy_check_mark:](./clients/js/src/features/wrap_type_map.ts) |  
+| Feature :heavy_check_mark: | Scenario :thought_balloon: | Required :question: | Spec :clipboard: | JS :scroll: | KT :scroll: |
+|-|-|-|-|-|-|
+| **`wrap://` URIs** | | | | | |
+| | Create & sanitize a `wrap://` URI | Yes | [:mag:](./specs/uri.yaml) | [:heavy_check_mark:](./clients/js/src/features/uri.ts) | [:heavy_check_mark:](./clients/kotlin/src/features/uri.ts) |
```
Comment (dOrgJelli) [inline README.md:9]: These checkmark links don't work, for example `./clients/kotlin/src/features/uri.ts` doesn't exist. Should be `./clients/kotlin/src/main/kotlin/features/uri/uri.kt`

File: clients/kotlin/src/main/kotlin/Main.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    val filter: String? = if (args.isNotEmpty() && args[0].isNotEmpty()) args[0] else null
+    val loader = SpecReader(root().resolve("specs"))
+
+    val specs: List<String> = listOf(
+        "config_embed_wrap_package",
+        "config_env_variables",
+        "config_interface_implementations",
+        "config_plugin_instance",
+        "config_plugin_package",
+//        "config_resolver",
```
Comment (dOrgJelli) [inline clients/kotlin/src/main/kotlin/Main.kt:20]: Any tests that are skipped should be marked with a :x: in the root README

File: clients/kotlin/src/main/kotlin/features/config/EmbedWrapPackage.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    val wasmModule = wrapDir.resolve("wrap.wasm").toFile().readBytes()
+
+    println("Creating WrapPackage from raw wrap.info & wrap.wasm bytes")
+
+    val wrapPackage = WasmPackage(manifest, wasmModule)
+
+    println("Adding WrapPackage to ClientConfig")
+
+    val client = polywrapClient {
+        setPackage("embed/foo" to wrapPackage)
```
Comment (dOrgJelli) [inline clients/kotlin/src/main/kotlin/features/config/EmbedWrapPackage.kt:38]: Very cool construction syntax :sunglasses:

**Review by dOrgJelli — CHANGES_REQUESTED** [review]
This looks amazing :clap: just some minor NITs around documenting the tests that aren't working, and fixing some broken links. Other than that this is fantastic and super exciting!

File: clients/kotlin/src/main/kotlin/Main.kt (hunk 2 above)
Comment (krisbitney) [reply clients/kotlin/src/main/kotlin/Main.kt:20]: Doesn't the ❌ imply the client can't do it? But really the client can add resolvers and can resolve URIs for invocations. Should we maybe have another shape to indicate this?

File: README.md (hunk 1 above)
Comment (krisbitney) [reply README.md]: oops! What a careless mistake. I fixed it.

File: clients/kotlin/src/main/kotlin/features/config/EmbedWrapPackage.kt (hunk 3 above)
Comment (krisbitney) [reply clients/kotlin/src/main/kotlin/features/config/EmbedWrapPackage.kt:38]: Thanks!

File: clients/kotlin/src/main/kotlin/Main.kt (hunk 2 above)
Comment (krisbitney) [reply clients/kotlin/src/main/kotlin/Main.kt:20]: I added the :x: next to all the skipped tests. The Kotlin client does support the config and URI resolution features that aren't being tested. It just doesn't support `tryResolveUri` or `loadWrapper`, which are required to complete the tests. 

If we could, for example, prove IPFS resolution by invoking a wrapper via IPFS, the Kotlin client would pass the test. I believe the same is true for all of the skipped tests except ENS resolution, which will be supported once we merge  https://github.com/polywrap/rust-client/pull/165

Comment (dOrgJelli) [reply clients/kotlin/src/main/kotlin/Main.kt:20]: Okay makes sense @krisbitney. Yah I think an orange circle 🟠 could work well for this. I’d also say we should update the test specs to not require tryResolveUri

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**